src/http/mod.rs

```diff
@@ -195,17 +196,17 @@ impl LightMethod {
 #[derive(Clone, Debug)]
 pub enum AttachmentType<'a> {
     /// Indicates that the `AttachmentType` is a byte slice with a filename.
-    Bytes((&'a [u8], &'a str)),
+    Bytes{ data: Cow<'a, [u8]>, filename: String } ,
     /// Indicates that the `AttachmentType` is a `File`
-    File((&'a File, &'a str)),
+    File{ file: &'a File, filename: String },
     /// Indicates that the `AttachmentType` is a `Path`
     Path(&'a Path),
     #[doc(hidden)]
     __Nonexhaustive,
 }
 
 impl<'a> From<(&'a [u8], &'a str)> for AttachmentType<'a> {
```

Comment:
Since we no longer depend on the filename's livability, its lifetime should be removed:
```suggestion
impl<'a> From<(&'a [u8], &str)> for AttachmentType<'a> {
```
This has the same effect as if we used two distinct lifetimes (`<'a, 'b>`) for the data (`'a`) and the filename (`'b`).